### ISSUE TITLE
HTML (& other junk) in Rendered Attributes

### ISSUE THREAD

**Issue by michaelsilver** [infolab-csail/WikipediaBase#42]
(I'm now using WikiBase as a library for the Ruby WikiBase backend, and I have been noticing bugs. I hope I am doing the right thing by reporting them here.)
For these, I should be able to regex-match and exlude these, but so you are aware:
```
mysql> select * from irendered where ren like '%<%';
+------------------------+---------------+----------------------------------------------+
| name                   | attr          | ren                                          |
+------------------------+---------------+----------------------------------------------+
| law enforcement agency | sub_div_type  | <span id="ibleaoj"/>Operations jurisdiction* |
| law enforcement agency | divname       | <span id="ibleaoj"/>Operations jurisdiction* |
| law enforcement agency | subdivtype    | <span id="ibleaoj"/>Operations jurisdiction* |
| law enforcement agency | div_type      | <span id="ibleaoj"/>Operations jurisdiction* |
| law enforcement agency | div_name      | <span id="ibleaoj"/>Operations jurisdiction* |
| law enforcement agency | divtype       | <span id="ibleaoj"/>Operations jurisdiction* |
| law enforcement agency | sub_div_name  | <span id="ibleaoj"/>Operations jurisdiction* |
| law enforcement agency | subdivname    | <span id="ibleaoj"/>Operations jurisdiction* |
| pharaoh                | golden        | <hiero>G16</hiero>                           |
| pharaoh                | burial        | <hiero>G8</hiero>                            |
| pharaoh                | children      | <hiero>G8</hiero>                            |
| pharaoh                | father        | <hiero>G8</hiero>                            |
| pharaoh                | monuments     | <hiero>G8</hiero>                            |
| pharaoh                | spouse        | <hiero>G8</hiero>                            |
| pharaoh                | death_date    | <hiero>G8</hiero>                            |
| pharaoh                | mother        | <hiero>G8</hiero>                            |
| pharaoh                | birth_date    | <hiero>G8</hiero>                            |
| station                | code          | Station code</th>                            |
| station                | bicycle       | Bicycle facilities</th>                      |
| station                | longs         | Coordinates</th>                             |
| station                | owned         | Owned by</th>                                |
| station                | connections   | Connections</th>                             |
| station                | longm         | Coordinates</th>                             |
| station                | parking       | Parking</th>                                 |
| station                | operator      | Operator</th>                                |
| station                | longEW        | Coordinates</th>                             |
| station                | opened        | Opened</th>                                  |
| station                | lats          | Coordinates</th>                             |
| station                | hangul        | Hangul</th>                                  |
| station                | platform      | Platforms</th>                               |
| station                | iso_region    | Coordinates</th>                             |
| station                | baggage_check | Baggage check</th>                           |
| station                | closed        | Closed</th>                                  |
| station                | latm          | Coordinates</th>                             |
| station                | rebuilt       | Rebuilt</th>                                 |
| station                | zone          | Fare zone</th>                               |
| station                | elevation     | Elevation</th>                               |
| station                | latNS         | Coordinates</th>                             |
| station                | electrified   | Electrified</th>                             |
| station                | latitude      | Coordinates</th>                             |
| station                | tracks        | Tracks</th>                                  |
| station                | levels        | Levels</th>                                  |
| station                | address       | Address</th>                                 |
| station                | line          | Line(s)</th>                                 |
| station                | former        | Formerly</th>                                |
| station                | structure     | Structure type</th>                          |
| station                | distance      | Distance</th>                                |
| station                | rr            | Revised Romanization</th>                    |
| station                | country       | Address</th>                                 |
| station                | hanja         | Hanja</th>                                   |
| station                | longitude     | Coordinates</th>                             |
| station                | depth         | Depth</th>                                   |
| station                | mr            | McCune-Reischauer</th>                       |
+------------------------+---------------+----------------------------------------------+
53 rows in set (0.00 sec)

mysql> select * from irendered where ren like '%&%';
+--------+------------------------+------------------+
| name   | attr                   | ren              |
+--------+------------------------+------------------+
| school | nursery_years_taught   | &mdot; Nursery   |
| school | secondary_years_taught | &mdot; Secondary |
| school | primary_years_taught   | &mdot; Primary   |
+--------+------------------------+------------------+
3 rows in set (0.00 sec)

mysql> select * from irendered where ren like '%{%';
+--------------------+--------+-------------------------------------------+
| name               | attr   | ren                                       |
+--------------------+--------+-------------------------------------------+
| artist discography | Option | [[#{{{Option name}}}|/]]{{{Option name}}} |
+--------------------+--------+-------------------------------------------+
1 row in set (0.00 sec)
```
Ideally, these would not have any HTML or other junk, and would just be human-readable text.

Note: This comes from only a small test subset of wikipedia (only including infoboxes from a random 1/100th subset of all articles). To reproduce these using calls to WikipediaBase:
```Python
>>> from wikipediabase.util import get_infobox, get_meta_infobox
>>> get_meta_infobox('Template:Infobox <name>').rendered_keys()
```
where `<name>` is whatever appears in the `name` column.

**Comment by fakedrake** [comment]
I am not sure what these tables mean... Are they from the ruby-wikibase?

**Comment by michaelsilver** [comment]
Yes they happen to be from Ruby WikipediBase's mysql backend, but all the data comes form your (Python) WikipediBase. I just used mysql's searching capabilities to pull up a list of funky attribute mappings. The first column is infobox name, second column is unrenedered, and third column is rendered. What I mean at the end is you can reproduce this yourself by running, for example
```Python
>>> from wikipediabase.util import get_infobox, get_meta_infobox
>>> get_meta_infobox('Template:Infobox law enforcement agency').rendered_keys()
```
and somewhere in the list of returned mappings, you will get the mappings I highlighted at the beginning of the first table.

**Comment by fakedrake** [comment]
Aha. Well it is probably a matter of the downloaded data being malformed. The python-wikibase properly parses all xml coming in (at the cost of some performance) so xml tags dont appear in the output unless they are meant to.

**Comment by michaelsilver** [comment]
Looks like PR #47 99.9% fixed this also. All the problems mentioned in the tables are now gone except for `u'Option': u'[[#{{{Option name}}}|\u2199]]{{{Option name}}}'` in `'Template:Infobox artist discography'`, but I think that one is more of an extreme-oddball than a general problem with HTML stuff appearing. Let's declare victory.

Closed by #47
